### ISSUE TITLE
0.5.1: pytest is failing in two units

### ISSUE THREAD

**Issue by kloczek** [mrbean-bremen/pytest-find-dependencies#7]
I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.1-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.1-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.1, configfile: setup.cfg, testpaths: tests
plugins: find-dependencies-0.5.1
collected 26 items

tests/test_find_dependencies.py ......................F..F                                                                                                           [100%]

================================================================================= FAILURES =================================================================================
__________________________________________________________________________ test_passed_arguments ___________________________________________________________________________

test_path = <Testdir local('/tmp/pytest-of-tkloczko/pytest-249/test_passed_arguments0')>

    def test_passed_arguments(test_path):
        test_path.makepyfile(
            test_one="""
            flag = True
            def test_a(): pass
            def test_b(): assert flag
            def test_c(): global flag; flag = False
            """
        )

        result = test_path.runpytest("--find-dependencies", "-p", "no:randomly",
                                     "-v", "-s")
        assert int(result.ret) == 1
>       result.stdout.fnmatch_lines([
            "Running pytest with arguments --find-dependencies-internal "
            "-n0 -p no:randomly -v -s *",
            "Run dependency analysis for 3 tests.",
            "Executed 7 tests in 3 test runs.",
            "Dependent tests:",
            "  test_one.py::test_b depends on test_one.py::test_c"
        ])
E       Failed: nomatch: 'Running pytest with arguments --find-dependencies-internal -n0 -p no:randomly -v -s *'
E           and: 'Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp'
E           and: 'Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp'
E           and: 'Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp'
E           and: ''
E           and: '============================== Results ==============================='
E           and: 'Run dependency analysis for 3 tests.'
E           and: 'Executed 7 tests in 3 test runs.'
E           and: 'Dependent tests:'
E           and: '  test_one.py::test_b depends on test_one.py::test_c'
E           and: ''
E           and: 'Dependency test FAILED'
E           and: '======================================================================'
E           and: ''
E       remains unmatched: 'Running pytest with arguments --find-dependencies-internal -n0 -p no:randomly -v -s *'

/home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.1/tests/test_find_dependencies.py:552: Failed
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp
Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp
Running pytest with arguments --find-dependencies-internal -p no:randomly -v -s --basetemp=/tmp/pytest-of-tkloczko/pytest-249/basetemp

============================== Results ===============================
Run dependency analysis for 3 tests.
Executed 7 tests in 3 test runs.
Dependent tests:
  test_one.py::test_b depends on test_one.py::test_c

Dependency test FAILED
======================================================================

_________________________________________________________________________ test_removed_xdist_args __________________________________________________________________________

test_path = <Testdir local('/tmp/pytest-of-tkloczko/pytest-249/test_removed_xdist_args0')>

    def test_removed_xdist_args(test_path):
        test_path.makepyfile(
            test_one="""
            def test_a(): print("one::a")
            def test_b(): print("one::b")
            """,
            test_two="""
            def test_c(): print("two::a")
            def test_d(): print("two::b")
            """
        )

        result = test_path.runpytest(
            "--find-dependencies", "-n", "2", "-s")
>       assert int(result.ret) == 0
E       assert 4 == 0
E        +  where 4 = int(<ExitCode.USAGE_ERROR: 4>)
E        +    where <ExitCode.USAGE_ERROR: 4> = <RunResult ret=ExitCode.USAGE_ERROR len(stdout.lines)=0 len(stderr.lines)=5 duration=0.08s>.ret

/home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.1/tests/test_find_dependencies.py:621: AssertionError
--------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: -n
  inifile: /tmp/pytest-of-tkloczko/pytest-249/test_removed_xdist_args0/pytest.ini
  rootdir: /tmp/pytest-of-tkloczko/pytest-249/test_removed_xdist_args0

========================================================================= short test summary info ==========================================================================
FAILED tests/test_find_dependencies.py::test_passed_arguments - Failed: nomatch: 'Running pytest with arguments --find-dependencies-internal -n0 -p no:randomly -v -s *'
FAILED tests/test_find_dependencies.py::test_removed_xdist_args - assert 4 == 0
====================================================================== 2 failed, 24 passed in 14.12s =======================================================================
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:79: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-8b14d20b-b28b-40f7-8e9e-5e6a9a8e815b
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-8b14d20b-b28b-40f7-8e9e-5e6a9a8e815b'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:79: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-eabe781e-27a0-4952-b5bc-f10555459b52
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-eabe781e-27a0-4952-b5bc-f10555459b52'
  warnings.warn(
```

**Comment by mrbean-bremen** [comment]
The tests assume that `pytest-xdist` is installed (they test the interaction with it). The tests in the CI are executed via tox, as is recommended for pytest plugins, and `xdist` is installed via `tox.ini`. I noticed though that the `requirements_dev.txt`, which is supposed to be used for tests otherwise, is missing `pytest-xdist`, though that may not help your use case. I'll probably just skip these tests in an environment without `pytest-xdist` installed.

**Comment by mrbean-bremen** [comment]
Shall be fixed in version 0.5.2.

**Comment by kloczek** [comment]
Tested and looks like it ok now
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.2-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-find-dependencies-0.5.2-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-find-dependencies-0.5.2, configfile: setup.cfg, testpaths: tests
plugins: find-dependencies-0.5.2
collected 26 items

tests/test_find_dependencies.py ......................s..s                                                                                                           [100%]

========================================================================= short test summary info ==========================================================================
SKIPPED [1] tests/test_find_dependencies.py:543: pytest-xdist not installed
SKIPPED [1] tests/test_find_dependencies.py:613: pytest-xdist not installed
====================================================================== 24 passed, 2 skipped in 12.89s ======================================================================
```
Thank you 😄
